spec.html

```diff
@@ -55,11 +55,7 @@ <h3>Basic Types</h3>
 
               <h3>Non-terminals</h3>
 
-              <p>The following specifies the rest of the BSON
-              grammar. Note that quoted strings represent terminals,
```

Comment:
As written, it looks like this patch continues to use the notation "\x00" for null terminals. It might make sense to define a terminal called "null_byte" as a shorthand for the 8 bit pattern 0000 0000? Another option would be to write it as signed_byte(0), although this might be confusing because whether or not this null byte should be interpreted as a signed number is irrelevant. Alternatively, you need to keep some version of the text here which defines the notation being used for terminals.